internal/querycoord/task.go

```diff
@@ -1933,49 +1934,48 @@ func assignInternalTask(ctx context.Context,
 	}
 	log.Debug("assignInternalTask: assign dmChannel to node success")
 
-	mergedLoadSegmentReqs := make(map[UniqueID]map[int64][]*querypb.LoadSegmentsRequest)
-	sizeCounts := make(map[UniqueID]map[int64]int)
-	for _, req := range loadSegmentRequests {
-		nodeID := req.DstNodeID
-		collectionID := req.CollectionID
-		if _, ok := mergedLoadSegmentReqs[collectionID]; !ok {
-			mergedLoadSegmentReqs[collectionID] = make(map[int64][]*querypb.LoadSegmentsRequest)
-			sizeCounts[collectionID] = make(map[int64]int)
-		}
-		mergedLoadSegmentReqsPerCol := mergedLoadSegmentReqs[collectionID]
-		sizeCountsPerCol := sizeCounts[collectionID]
-		sizeOfReq := getSizeOfLoadSegmentReq(req)
-		if _, ok := mergedLoadSegmentReqsPerCol[nodeID]; !ok {
-			mergedLoadSegmentReqsPerCol[nodeID] = []*querypb.LoadSegmentsRequest{}
-			mergedLoadSegmentReqsPerCol[nodeID] = append(mergedLoadSegmentReqsPerCol[nodeID], req)
-			sizeCountsPerCol[nodeID] = sizeOfReq
-		} else {
-			if sizeCountsPerCol[nodeID]+sizeOfReq > MaxSendSizeToEtcd {
-				mergedLoadSegmentReqsPerCol[nodeID] = append(mergedLoadSegmentReqsPerCol[nodeID], req)
-				sizeCountsPerCol[nodeID] = sizeOfReq
-			} else {
-				lastReq := mergedLoadSegmentReqsPerCol[nodeID][len(mergedLoadSegmentReqsPerCol[nodeID])-1]
-				lastReq.Infos = append(lastReq.Infos, req.Infos...)
-				sizeCountsPerCol[nodeID] += sizeOfReq
-			}
-		}
-	}
+	if len(loadSegmentRequests) > 0 {
+		sort.Slice(loadSegmentRequests, func(i, j int) bool {
+			return loadSegmentRequests[i].CollectionID < loadSegmentRequests[j].CollectionID ||
+				loadSegmentRequests[i].CollectionID == loadSegmentRequests[j].CollectionID && loadSegmentRequests[i].DstNodeID < loadSegmentRequests[j].DstNodeID
+		})
 
-	for _, loadSegmentsReqsPerCol := range mergedLoadSegmentReqs {
-		for _, loadSegmentReqs := range loadSegmentsReqsPerCol {
-			for _, req := range loadSegmentReqs {
+		batchReq := loadSegmentRequests[0]
+		batchSize := proto.Size(batchReq)
```

Comment:
Maybe using `getSizeOfLoadSegmentReq` will be safer if we want to change the way of estimating size of protobuf serialization(which might not happen). This function is simple and it would be inlined.

Reply:
Actually we could choose to delete this util function if it's not used anymore.

Reply:
prefer to remove it